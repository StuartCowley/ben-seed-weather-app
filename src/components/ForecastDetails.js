import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__maxTemperature">
        Max Temperature: {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__minTemperature">
        Min Temperature: {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity} %</div>
      <div className="forecast-details__windSpeed">
        Wind Speed: {wind.speed} mph
      </div>
      <div className="forecast-details__windDirection">
        Wind Direction: {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
