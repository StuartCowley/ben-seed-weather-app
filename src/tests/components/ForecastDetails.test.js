import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1525046400000,
    humidity: 10,
    wind: {
      speed: 100,
      direction: "South East",
    },
    temperature: {
      max: 11,
      min: 2,
    },
  };

  describe("ForecastDetails-snapshot", () => {
    it("renders props correctly", () => {
      const { asFragment } = render(
        <ForecastDetails forecast={validProps} /> // see line 7 of ForecastDetails.js
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Mon Apr 30 2018")).toHaveAttribute(
      "class",
      "forecast-details__date"
    );
    expect(getByText("Max Temperature: 11°C")).toHaveAttribute(
      "class",
      "forecast-details__maxTemperature"
    );
    expect(getByText("Min Temperature: 2°C")).toHaveAttribute(
      "class",
      "forecast-details__minTemperature"
    );
    expect(getByText("Humidity: 10")).toHaveAttribute(
      "class",
      "forecast-details__humidity"
    );
    expect(getByText("Wind Speed: 100")).toHaveAttribute(
      "class",
      "forecast-details__windSpeed"
    );
    expect(getByText("Wind Direction: South East")).toHaveAttribute(
      "class",
      "forecast-details__windDirection"
    );
  });
});
