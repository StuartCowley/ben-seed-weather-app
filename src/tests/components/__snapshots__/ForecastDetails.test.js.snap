// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails ForecastDetails-snapshot renders props correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__maxTemperature"
    >
      Max Temperature: 11°C
    </div>
    <div
      class="forecast-details__minTemperature"
    >
      Min Temperature: 2°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 10
    </div>
    <div
      class="forecast-details__windSpeed"
    >
      Wind Speed: 100
    </div>
    <div
      class="forecast-details__windDirection"
    >
      Wind Direction: South East
    </div>
  </div>
</DocumentFragment>
`;
